File: protocol-generator/java/src/main/java/org/entitycore/protocol/codec/TagRejectedException.java
package org.entitycore.protocol.codec;

/** A CBOR major-type-6 tag was encountered at any depth on decode. ECF forbids
 *  tags entirely (invariant N2 / §6.3) — hard reject, no recovery. Status 400. */
public class TagRejectedException extends EntityCodecException {
    private static final long serialVersionUID = 1L;
    public TagRejectedException(String message) { super(message); }
}

File: protocol-generator/php/src/TagRejectedException.php
<?php

declare(strict_types=1);

namespace EntityCore;

/**
 * A CBOR tag (major type 6) was found in the input. ECF §6.3 / invariant N2:
 * tags MUST be rejected at ANY nesting depth → 400 non_canonical_ecf.
 *
 * A subclass of NonCanonicalEcfException (a tag is one species of non-canonical
 * input) so a `catch (NonCanonicalEcfException)` catches it too.
 */
class TagRejectedException extends NonCanonicalEcfException
{
}
